infra/experimental/sanitizers/ExecSan/execSan.cpp

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright 2022 Google LLC
+
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+
+ *      http://www.apache.org/licenses/LICENSE-2.0
+
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* A detector that uses ptrace to identify shell injection vulnerabilities. */
+
+/* C standard library */
+#include <errno.h>
+#include <stdio.h>
+#include <string.h>
+#include <signal.h>
+
+/* POSIX */
+#include <unistd.h>
+#include <sys/stat.h>
+#include <sys/user.h>
+#include <sys/wait.h>
+
+/* Linux */
+#include <syscall.h>
+#include <sys/ptrace.h>
+
+#include <fstream>
+#include <string>
+#include <vector>
+#include <set>
+
+const std::string kTripWire = "/tmp/tripwire";
+
+#define DEBUG_LOGS 0
+
+#if DEBUG_LOGS
+#define debug_log(...) \
+  do { \
+    fprintf(stderr, __VA_ARGS__); fflush(stdout); \
+    fputc('\n', stderr); \
+  } while (0)
+#else
+#define debug_log(...)
+#endif
+
+#define fatal_log(...) \
+  do { \
+    fprintf(stderr, __VA_ARGS__); \
+    fputc('\n', stderr); \
+    exit(EXIT_FAILURE); \
+  } while (0)
+
+pid_t run_child(char **argv) {
```

Comment:
Don't think it's your doing but it's bad that this function is supposed to take this program's argv and not argv to pass to the child to execute.